# Session 5 Solution: GitLab CI Integration (Entra Auth)

In [1]:
from pathlib import Path

template_path = Path.cwd().parent / "templates" / "gitlab-ci.yml"
if not template_path.exists():
    template_path = Path.cwd() / "solutions" / "templates" / "gitlab-ci.yml"

print(template_path.read_text(encoding="utf-8"))

stages:
  - evaluate

variables:
  PIP_DISABLE_PIP_VERSION_CHECK: "1"
  PYTHONUNBUFFERED: "1"
  MIN_COHERENCE: "0.85"
  MIN_RELEVANCE: "0.85"

evaluate_agent:
  image: python:3.11
  stage: evaluate
  script:
    - python -m venv .venv
    - . .venv/bin/activate
    - pip install --upgrade pip
    - pip install -r solutions/requirements.txt
    - python solutions/scripts/run_cloud_evaluation.py
  only:
    - main

# Required protected CI/CD variables for Entra service principal auth:
# - PROJECT_ENDPOINT
# - MODEL_DEPLOYMENT_NAME
# - AZURE_TENANT_ID
# - AZURE_CLIENT_ID
# - AZURE_CLIENT_SECRET



## Required GitLab CI/CD Variables (Entra Service Principal)
- PROJECT_ENDPOINT
- MODEL_DEPLOYMENT_NAME
- AZURE_TENANT_ID
- AZURE_CLIENT_ID
- AZURE_CLIENT_SECRET
- MIN_COHERENCE (optional, default 0.85)
- MIN_RELEVANCE (optional, default 0.85)

The CI job runs `python solutions/scripts/run_cloud_evaluation.py`, which calls:
- `openai_client.evals.create(...)`
- `openai_client.evals.runs.create(...)`
- `openai_client.evals.runs.retrieve(...)` until completion